CGPs/cgp-0056.md

```diff
@@ -0,0 +1,57 @@
+---
+cgp: 56
+title: Adding Di Wu as oracle provider
+date-created: 2022-06-14
+author: Martin Volpe (@martinvol, volpe@clabs.co)
+status: DRAFT
+discussions-to: TBD
+governance-proposal-id: [if submitted]
+date-executed: [if executed] <date created on, in ISO 8601 (yyyy-mm-dd) format>
+---
+## Overview
+
+The proposal is to allow Di Wu as an oracle runner for [Mento](https://docs.celo.org/celo-codebase/protocol/stability/doto#what-is-mento), the stability mechanism behind the Celo Stables. Di Wu has previously showed interesting in running the oracles in a [forum post]([url](https://github.com/celo-org/governance/issues/142)) calling the community to run oracles.
+
+Before submitting the proposal on-chain, the community will asses the security setup of the new oracle clients, and request a signature from Di Wu to validate the addresses are indeed his.
+
+Di Wu has offered to provide this service ad honorem.
+
+## Proposed Changes
+
+1. Enable Di Wi CELO/USD oracle
+  - Destination: Enabling one address(TBD) to report to CELO/USD
+  - Data: `SortedOracles.addOracle(cUSD address, ADDRESS1-DIWU)`
+  - Value: 0
+2. Enable Di Wi CELO/EUR oracle
+  - Destination: Enabling one address(TBD) to report to CELO/EUR
+  - Data: `SortedOracles.addOracle(cEUR address, ADDRESS2-DIWU)`
+  - Value: 0
+3. Enable Di Wi CELO/BRL oracle
+  - Destination: Enabling one address(TBD) to report to CELO/BRL
+  - Data: `SortedOracles.addOracle(cREAL address, ADDRESS3-DIWU)`
+  - Value: 0
+4. Disable cLabs CELO/USD oracle
+  - Destination: Disable one address(TBD) to report to CELO/USD
+  - Data: `SortedOracles.removeOracle(cUSD address, ADDRESS1-CLABS)`
+  - Value: 0
+5. Disable cLabs CELO/EUR oracle
+  - Destination: Disable one address(TBD) to report to CELO/EUR
+  - Data: `SortedOracles.removeOracle(cEUR address, ADDRESS2-CLABS)`
+  - Value: 0
+6. Disable cLabs CELO/BRL oracle
+  - Destination: Disable one address(TBD) to report to CELO/BRL
+  - Data: `SortedOracles.removeOracle(cREAL address, ADDRESS3-CLABS)`
+  - Value: 0
+
+## Verification
+
+`celocli governance:view --proposalID TODO`
+
+## Risks
+
+cLabs oracles work by enabling a list of oracles, enable them to report and calculating a median of the report. One individual provider can not influence the median on their own, the only way to affect it would be to control half the oracles plus one.
```

Comment:
```suggestion
Celo oracles work by enabling a list of oracles, enable them to report and calculating a median of the report. One individual provider can not influence the median on their own, the only way to affect it would be to control half the oracles plus one.
```